packages/crayons-icon/icon-utils/icon-registry.js

```diff
@@ -0,0 +1,43 @@
+class IconRegistry {
+  constructor() {
+    if (typeof window !== 'undefined') {
+      this.iconLibRegistry = window.sessionStorage;
```

Comment:
Instead of using session storage can we look at using global functions/controllers ?, Using a window object can be a problem when mfe apps use crayons. There could be conflicts in the name of the lib being used and they can get overridden by mistake.

Reply:
As discussed , this is okay implementation from optimisation perspective. We can plan other approaches later as its not very important. 
Singletons/Global Functions exist at the process level. This means that if you have 20 users accessing your site at any given moment, they're using the same singleton object.Sessions exist at the user level. That means you can store data per user, not per process. Developers would register 1/2 lib at max and we are only storing path info. So keeping [key:value] pair in window object gives us a good way to make it accesible to all users and across pages. Also if you refresh the page your globals are maintained. In case of singletons it gets destroyed and recreated. Also doesn't works always. We also have to write a global function to handle such registrations and make it available across components which is maintenance overhead. window.sessionStorage is safe and gives us good apis to accomplish our task. We can proper namespace our libs to avoid collision in decimal cases.
https://stackoverflow.com/questions/2821941/session-vs-singleton-pattern.

Reply:
I don't think if 20 users are accessing the site, they would be sharing the singleton object. This will be per consumer of the crayons. Each consumer of the crayons should get an instance of IconRegistry and it won't be shared b/w the consumers so that this won't be a problem with MFE. But if we go with window and session storage approach, We have to inform the consumers to namespace the keys when they register with the icon lib to avoid values being overridden.

@parsuram-vijaysankar kindly share your views on this.

Reply:
Good that you brought the point of an MFE scenario... 
Yes, if two apps use the same name, by mistake, that will be hard to debug the problem. 

@detiwari003 Let's go with the singleton approach in this case... Let's remember that always configurations need to be specific to MFEs and a shared state would just make it a bad architecture!

Reply:
I tried to follow and implemented this model.
[](https://codesandbox.io/s/elegant-swanson-i0g07).
This doesn't works in Crayons system and renders a blank object. Copied same module import concept to registry for state persistence but doesn't works for stencil web components.This same approach is used in Shoelace and seems to works till you don't refresh the page. 
We have no choice but go with window.sessionStorage as in PR . Singleton module imports to access lib registrations in icon component doesn't work because of following reasons:-
1. We are using these imports in .tsx files which are in turn managed by Stencil Compiler. So all its relative imports don't seem to follow JS Module import system and creates new instance each time. Shoelace is using icon.ts file and not icon.tsx so module imports persist across boundaries when setting data via setters.
2. Also the data if shared via singleton across modules seems to persist for js files but for UI States this resets on page refresh. This problem Angular also has even if they are using DI and Injector service. Developers have used localStorage to persist the data on page refresh. Threads available on stackoverflow.